lib/core/logconf.c

```diff
@@ -289,6 +289,12 @@ h2o_logconf_t *h2o_logconf_compile(const char *fmt, int escape, char *errbuf)
                     MAP_EXT_TO_PROTO("ssl.cipher-bits", ssl.cipher_bits);
                     MAP_EXT_TO_PROTO("ssl.session-id", ssl.session_id);
                     MAP_EXT_TO_PROTO("ssl.server-name", ssl.server_name);
+                    MAP_EXT_TO_PROTO("proxy.ssl.protocol-version", proxy_ssl.protocol_version);
+                    MAP_EXT_TO_PROTO("proxy.ssl.session-reused", proxy_ssl.session_reused);
+                    MAP_EXT_TO_PROTO("proxy.ssl.cipher", proxy_ssl.cipher);
+                    MAP_EXT_TO_PROTO("proxy.ssl.cipher-bits", proxy_ssl.cipher_bits);
+                    MAP_EXT_TO_PROTO("proxy.ssl.session-id", proxy_ssl.session_id);
+                    MAP_EXT_TO_PROTO("proxy.ssl.server-name", proxy_ssl.server_name);
```

Comment:
Maybe you missed removing some from here?

Reply:
My intention was to leave format string and conn_callbacks declaration as now so that
when user using the format, h2o output "-" since the correspnding callbacks of http1/http2 conn have NULL values.

When we separate downstream and upstream SSL info declaration, we cannot use like the following:
```
            struct {
                h2o_iovec_t (*protocol_version)(h2o_req_t *req);
                h2o_iovec_t (*session_reused)(h2o_req_t *req);
                h2o_iovec_t (*cipher)(h2o_req_t *req);
                h2o_iovec_t (*cipher_bits)(h2o_req_t *req);
                h2o_iovec_t (*session_id)(h2o_req_t *req);
                h2o_iovec_t (*server_name)(h2o_req_t *req);
            } ssl, proxy_ssl; /* upstream proxy ssl */
```
but use the following declaration and I prefered the former decl:
```
            struct {
                h2o_iovec_t (*protocol_version)(h2o_req_t *req);
                h2o_iovec_t (*session_reused)(h2o_req_t *req);
                h2o_iovec_t (*cipher)(h2o_req_t *req);
                h2o_iovec_t (*cipher_bits)(h2o_req_t *req);
                h2o_iovec_t (*session_id)(h2o_req_t *req);
                h2o_iovec_t (*server_name)(h2o_req_t *req);
            } ssl;
            struct {
                h2o_iovec_t (*protocol_version)(h2o_req_t *req);
                h2o_iovec_t (*session_reused)(h2o_req_t *req);
                h2o_iovec_t (*cipher)(h2o_req_t *req);
                h2o_iovec_t (*cipher_bits)(h2o_req_t *req);
            } proxy_ssl;
```
Which do you prefer?

Reply:
Ah I see, thank you for the clarification.

Rereading the code, I tend to think that the new attributes that we are trying to add (i.e. `proxy.ssl.*`) should be defined like `ELEMENT_TYPE_PROXY_IDLE_TIME`, meaning that the mapping would be defined using `MAP_EXT_TO_TYPE`, and we would have a `case` statement for each of the attributes.

The added properties do not depend on the protocol being used for talking to the client, and therefore do not have to have callback entries in each of the protocol implementation.

Reply:
Thank you for your kind comment.
I totally agree with you. I've posted the f/u pull request.